smoke-test/src/test/java/org/opennms/smoketest/minion/TelemetryIT.java

```diff
@@ -0,0 +1,340 @@
+import org.junit.Before;
+import org.junit.ClassRule;
+import org.junit.Test;
+import org.opennms.core.criteria.Criteria;
+import org.opennms.core.criteria.CriteriaBuilder;
+import org.opennms.core.xml.JaxbUtils;
+import org.opennms.netmgt.dao.api.EventDao;
+import org.opennms.netmgt.dao.api.NodeDao;
+import org.opennms.netmgt.dao.hibernate.EventDaoHibernate;
+import org.opennms.netmgt.dao.hibernate.NodeDaoHibernate;
+import org.opennms.netmgt.events.api.EventConstants;
+import org.opennms.netmgt.model.OnmsEvent;
+import org.opennms.netmgt.model.OnmsNode;
+import org.opennms.netmgt.telemetry.adapters.jti.proto.Port;
+import org.opennms.netmgt.telemetry.adapters.jti.proto.TelemetryTop;
+import org.opennms.netmgt.xml.event.Event;
+import org.opennms.netmgt.xml.event.Parm;
+import org.opennms.netmgt.xml.event.Value;
+import org.opennms.smoketest.NullTestEnvironment;
+import org.opennms.smoketest.OpenNMSSeleniumTestCase;
+import org.opennms.smoketest.utils.DaoUtils;
+import org.opennms.smoketest.utils.HibernateDaoFactory;
+import org.opennms.test.system.api.NewTestEnvironment.ContainerAlias;
+import org.opennms.test.system.api.utils.SshClient;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.opennms.test.system.api.TestEnvironment;
+import org.opennms.test.system.api.TestEnvironmentBuilder;
+
+/**
+ * Verifies that Telemetry listeners can receive proto buffers and generate rrd
+ * files
+ *
+ * @author cgorantla
+ */
+
+public class TelemetryIT {
+
+    private static final Logger LOG = LoggerFactory.getLogger(TelemetryIT.class);
+    public static final String SENDER_IP = "192.168.1.1";
+    public static final String SENDER_IP_MINION = "192.168.1.2";
+
+    private static TestEnvironment m_testEnvironment;
+
+    @ClassRule
+    public static final TestEnvironment getTestEnvironment() {
+        if (!OpenNMSSeleniumTestCase.isDockerEnabled()) {
+            return new NullTestEnvironment();
+        }
+        try {
+            final TestEnvironmentBuilder builder = TestEnvironment.builder().all();
+            builder.withOpenNMSEnvironment()
+                    .addFile(TelemetryIT.class.getResource("/telemetryd-configuration.xml"),
+                            "etc/telemetryd-configuration.xml")
+                    .addFile(TelemetryIT.class.getResource("/telemetryd-adapters/junos-telemetry-interface.groovy"),
+                            "etc/telemetryd-adapters/junos-telemetry-interface.groovy");
+            OpenNMSSeleniumTestCase.configureTestEnvironment(builder);
+            m_testEnvironment = builder.build();
+            return m_testEnvironment;
+        } catch (final Throwable t) {
+            throw new RuntimeException(t);
+        }
+    }
+
+    @Before
+    public void checkForDocker() {
+        Assume.assumeTrue(OpenNMSSeleniumTestCase.isDockerEnabled());
+    }
+
+    @Test
+    public void verifyJtiListenerOnOpenNMS() throws IOException, InterruptedException, JAXBException {
```

Comment:
Most of the code in the Minion and OpenNMS tests is the same. Can we refactor this to share the common blocks?